service/frontend/wrappers/clusterredirection/policy.go

```diff
@@ -132,6 +132,14 @@ var selectedAPIsForwardingRedirectionPolicyAPIAllowlistV2 = map[string]struct{}{
 	"RespondActivityTaskFailedByID":    {},
 }
 
+// allowedAPIsForDeprecatedDomains contains a list of APIs that are allowed to be called on deprecated domains
+var allowedAPIsForDeprecatedDomains = map[string]struct{}{
```

Comment:
This check could be moved to the template itself.
See service/frontend/templates/clusterredirection.tmpl and usages like $nonFowradingAPIs
So domainEntry.IsDeprecatedOrDeleted() will be added only if it required.

Reply:
Well, it could be done with a template, but it is not a bottleneck here. So it should be fine to just use a map here.